new and improved sql joins demystified (#272)
arcus/education_modules: sql_joins_demystified/sql_joins_demystified.md

<!--
author:   Joy Payton
email:    paytonk@chop.edu
version:  1.0.0
module_template_version: 3.0.0
language: en
narrator: US English Female
title: SQL Joins Demystified
comment: Learn about SQL joins in a non-coding module.

> long_description: Usually, data in a SQL database is organized into multiple interrelated tables.  This means you will often have to bring data together from two or more tables into a single dataset to answer your research questions.  This is accomplished using JOIN commands.  This module explains the type of joins, without going into detail about the SQL code itself.

Reviewer: ```suggestion
long_description: Usually, data in a SQL database is organized into multiple interrelated tables.  This means you will often have to bring data together from two or more tables into a single dataset to answer your research questions.  This is accomplished using JOIN commands.  This module explains the types of joins, without going into detail about the SQL code itself.
```

estimated_time: 1 hour

@learning_objectives  
After completion of this module, learners will be able to:

> - Explain what "join criteria" are and give an example of what makes "matching" data in two separate tables

Reviewer: ```suggestion
- Explain what "join criteria" are
- Give an example of what makes "matching" data in two separate tables
```

</div>
****

## Additional Resources

* Want to understand the basics of boolean algebra?  Check out a good tutorial at https://ryanstutorials.net/boolean-algebra-tutorial/boolean-algebra.php.
* The second part of that tutorial covers some of the rules (like commutativity) of boolean algebra: https://ryanstutorials.net/boolean-algebra-tutorial/boolean-algebra-laws.php 
* Finally, to understand the order of operations in boolean algebra a bit more intuitively, we recommend the third part of the same tutorial: https://ryanstutorials.net/boolean-algebra-tutorial/boolean-algebra-expressions.php 
* A great page to practice joins and visualize them in the form of Venn diagrams is at https://www.w3schools.com/sql/sql_join.asp

## Feedback

In the beginning, we stated some goals.

**Learning Objectives:**

@learning_objectives

We ask you to fill out a brief (5 minutes or less) survey to let us know:

* If we achieved the learning objectives
* If the module difficulty was appropriate
* If we gave you the experience you expected

> We gather this information in order to iteratively improve our work.  Thank you in advance for filling out [our brief survey](https://redcap.chop.edu/surveys/?s=KHTXCXJJ93&module_name=%22SQL+Joins%22Demystified)!

Reviewer: ```suggestion
We gather this information in order to iteratively improve our work.  Thank you in advance for filling out [our brief survey](https://redcap.chop.edu/surveys/?s=KHTXCXJJ93&module_name=%22SQL+Joins+Demystified%22)!
```
